lib/bundler/definition.rb

```diff
@@ -644,7 +644,8 @@ def converge_sources
       if !locked_gem_sources.empty? && !actual_remotes.empty?
         locked_gem_sources.each do |locked_gem|
           # Merge the remotes from the Gemfile into the Gemfile.lock
-          changes |= locked_gem.replace_remotes(actual_remotes)
+          changes |= !locked_gem.equivalent_remotes?(actual_remotes)
```

Comment:
I was thinking to replace the hosts but still return false if none of them have changed (besides the auth info)